rotkehlchen/chain/evm/decoding/pendle/decoder.py

```diff
@@ -390,11 +416,109 @@ def _decode_exit_post_exp_to_token(self, context: DecoderContext) -> DecodingOut
+                break
+        else:
+            log.error(f'Could not find the pendle claim transfer for transaction {context.transaction}')  # noqa: E501
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_redeem_interests(self, context: DecoderContext) -> DecodingOutput:
+        """Decode a Pendle redeem interests transaction."""
+        if context.tx_log.topics[0] != REDEEM_TOPIC:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount_in = asset_normalized_value(
+            asset=(token_in := self.base.get_token_or_native(bytes_to_address(context.tx_log.topics[3]))),  # noqa: E501
+            amount=int.from_bytes(context.tx_log.data[:32]),
+        )
+        amount_out = asset_normalized_value(
+            asset=(token_out := self.base.get_or_create_evm_token(context.tx_log.address)),
+            amount=int.from_bytes(context.tx_log.data[32:64]),
+        )
+        sy_in_event, sy_out_event = None, None
+        for event in context.decoded_events:
+            if event.asset == token_out and event.amount == amount_out:
+                if (
+                        event.event_type == HistoryEventType.RECEIVE and
+                        event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    sy_in_event = event
+
+                elif (
+                        event.event_type == HistoryEventType.SPEND and
+                        event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    sy_out_event = event
+
+            elif (
+                    event.asset == token_in and
+                    event.amount == amount_in and
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+            ):
+                event.counterparty = CPT_PENDLE
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.notes = f'Withdraw {event.amount} {token_in.symbol} from Pendle'
+
+        if not sy_in_event or not sy_out_event:
+            log.error(f'Failed to in & out events of sy token for Pendle redeem interests transaction {context.transaction}')  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        # these SY receive/spend events cancel each other out, so we remove them.
+        context.decoded_events.remove(sy_in_event)
+        context.decoded_events.remove(sy_out_event)
+        return DEFAULT_DECODING_OUTPUT
+
+    def reload_data(self) -> Mapping['ChecksumEvmAddress', tuple[Any, ...]] | None:
+        if should_update_protocol_cache(
+            cache_key=CacheType.PENDLE_POOLS,
+            args=(str(self.evm_inquirer.chain_id.serialize()),),
+        ) is True or should_update_protocol_cache(
+            cache_key=CacheType.PENDLE_SY_TOKENS,
+            args=(str(self.evm_inquirer.chain_id.serialize()),),
+        ) is True:
```

Comment:
nitpick for boolean checks 

```suggestion
        ):
```